bib/internal/setup/setup.go

```diff
@@ -0,0 +1,98 @@
+package setup
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/osbuild/bootc-image-builder/bib/internal/podmanutils"
+	"github.com/osbuild/bootc-image-builder/bib/internal/utils"
+	"golang.org/x/sys/unix"
+)
+
+// EnsureEnvironment mutates external filesystem state as necessary
+// to run in a container environment.  This function is idempotent.
+func EnsureEnvironment() error {
+	osbuildPath := "/usr/bin/osbuild"
+	if utils.IsMountpoint(osbuildPath) {
+		return nil
+	}
+
+	// A hardcoded security label from Fedora derivatives
+	// TODO: Avoid hardcoding this by using either host policy lookup
+	// Or eventually depend on privileged containers just having this capability.
+	//
+	// We need this in order to get `install_t` that has `CAP_MAC_ADMIN` for creating SELinux
+	// labels unknown to the host.
+	//
+	// Note that the transition to `install_t` must happen at this point. Osbuild stages run in `bwrap` that creates
+	// a nosuid, no_new_privs environment. In such an environment, we cannot transition from `unconfined_t` to `install_t`,
+	// because we would get more privileges.
+	installType := "system_u:object_r:install_exec_t:s0"
+	// Where we dump temporary files; this must be an overlayfs as we cannot
+	// write security contexts on overlayfs.
+	runTmp := "/run/osbuild/"
+
+	if err := os.MkdirAll(runTmp, 0o755); err != nil {
+		return err
+	}
+	if !utils.IsMountpoint(runTmp) {
+		if err := exec.Command("mount", "-t", "tmpfs", "tmpfs", runTmp).Run(); err != nil {
+			return fmt.Errorf("failed to mount tmpfs to %s: %w", runTmp, err)
+		}
+	}
+	src, err := os.Open("/usr/bin/osbuild")
```

Comment:
(nipitkc) I wonder if just `exec.Command("cp",...` might be easier here as it's shorter ?